### PR TITLE
Feat/123 receipient feedback firebase calls

### LINKED ISSUE #390: Recipients can accept missions and express gratitude.
**Description**

As a recipient, I can accept a mission and express my gratitude so everyone knows it was a success.

[Airtable #123](https://airtable.com/tblRjd2McKgRW9MKj/viwlRRNI0PZIxrGkc/rec6Af58cTPFl1Fpg?blocks=hide)

[Figma](https://www.figma.com/file/GLg0zPNnwMEGIb9TytCrxQ/Resilience-App?node-id=2527%3A4277)

### PR DESCRIPTION
**Pull request by amanchauhan11**
Fixes #390 

[Airtable](https://airtable.com/tblRjd2McKgRW9MKj/viwlRRNI0PZIxrGkc/rec6Af58cTPFl1Fpg)
In continuation of this [PR](https://github.com/factn/resilience-app/pull/470)
Added methods in `Mission.ts` to submit feedback. Implemented submit handlers in UI.

TODOs
1. Firestore rules to make sure only recipient of that mission can submit feedback and mission status is delivered.
2. Unit tests.

### CODE REVIEW

File: src/app/page/MissionFeedback/Steps.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+}) => {
+  const classes = useStyles();
+  const [error, setError] = useState(false);
+  const onSubmit = () => {
+    if (required && !value) {
+      setError(true);
+    } else {
+      handleSubmit();
+    }
+  };
```
Comment (amanchauhan11) [inline src/app/page/MissionFeedback/Steps.js:88]: Is this OK for checking empty input? The form just had one text field so kept it simple.

File: src/app/page/MissionsCreated/Missions.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import PropTypes from "prop-types";
```
Comment (qdozaq) [inline src/app/page/MissionsCreated/Missions.jsx:1]: As a heads up I'm reworking this page in `/src/app/page/Recipient/Dashboard.jsx`

File: scheme/data.json [hunk 3]
```diff
@@ -3,3488 +3,3488 @@
     "1": {
       "missions": {
         "669230ba45838b4a673561796cdb0e02": {
-          "status": "started", 
+          "status": "started",
```
[the remark this answers is not in the document]
Comment (amanchauhan11) [reply scheme/data.json]: Fixed  in `446e54eda87ebb0f7df47e69c66ea5320b616c87`.

File: src/app/model/Mission.ts [hunk 4]
```diff
@@ -350,6 +350,32 @@ class Mission extends BaseModel {
       status: MissionStatus.tentative,
     });
   }
+
+  /**
+   * Submit feedback for mission
+   * @param {string} missionUid : mission for which feedback will be set
+   * @param {string} feedback: mission feedback
+   */
+  submitFeedback(missionUid: string, feedback: string) {
```
[the remark this answers is not in the document]
Comment (amanchauhan11) [reply src/app/model/Mission.ts]: Fixed in this [commit](https://github.com/factn/resilience-app/pull/506/commits/446e54eda87ebb0f7df47e69c66ea5320b616c87)

File: src/app/page/MissionsCreated/Missions.jsx (hunk 2 above)
Comment (amanchauhan11) [reply src/app/page/MissionsCreated/Missions.jsx:1]: Yeah. I saw that one. Just add the link `/missions/feedback/:uid` to submit feedback.

File: src/app/model/Mission.ts [hunk 5]
```diff
@@ -384,6 +384,21 @@ class Mission extends BaseModel {
...
+   * Submit feedback for mission
+   * @param {string} missionUid : mission for which feedback will be set
+   * @param {string} feedback: mission feedback
+   * @param {boolean} success: whether the the mission was a success.
+   */
+  submitFeedback(missionUid: string, feedback: string, success: boolean) {
+    //TODO: update firestore.rules to allow only if the mission is created by that user
+    return this.update(missionUid, {
+      feedbackNotes: feedback || "",
+      ...(success && { status: MissionStatus.succeeded }),
```
Comment (amanchauhan11) [inline src/app/model/Mission.ts:398]: Let me know if this short-hand seems hacky. Found it [here](https://stackoverflow.com/questions/50367085/assign-and-set-js-object-property-conditionally)

Comment (qdozaq) [reply src/app/model/Mission.ts:398]: interesting. A little unorthodox, but I think it's fine for this PR.

**Pull request merged**